File: frontend/src/types/recommendations.ts
export type RecommendationReason = 'weak_mastery' | 'try_something_new' | 'challenge'

export interface RecommendationEntry {
  category: string | null
  reason: RecommendationReason
  message: string
  mastery: number | null
  action_label: string
  action_path: string
}

export interface RecommendationsResponse {
  recommendations: RecommendationEntry[]
}

File: backend/app/schemas/recommendations.py
from typing import Literal

from pydantic import BaseModel

RecommendationReason = Literal["weak_mastery", "try_something_new", "challenge"]


class RecommendationEntry(BaseModel):
    category: str | None
    reason: RecommendationReason
    message: str
    mastery: float | None
    action_label: str
    action_path: str


class RecommendationsResponse(BaseModel):
    recommendations: list[RecommendationEntry]
